crates/noirc_evaluator/src/ssa_refactor/ir/dom.rs

```diff
@@ -0,0 +1,426 @@
+//! The dominator tree of a function, represented as a hash map of each reachable block id to its
+//! immediate dominator.
+//!
+//! Dominator trees are useful for tasks such as identifying back-edges in loop analysis or
+//! calculating dominance frontiers.
+
+use std::{cmp::Ordering, collections::HashMap};
+
+use super::{basic_block::BasicBlockId, cfg::ControlFlowGraph, post_order::PostOrder};
+
+/// Dominator tree node. We keep one of these per reachable block.
+#[derive(Clone, Default)]
+struct DomNode {
+    /// The block's idx in the control flow graph's reverse post-order
+    reverse_post_order_idx: u32,
+
+    /// The block that immediately dominated that of the node in question.
+    ///
+    /// This will be None for the entry block, which has no immediate dominator.
+    immediate_dominator: Option<BasicBlockId>,
+}
+
+impl DomNode {
```

Comment:
Since we're spelling out DominatorTree I think we should spell out DominatorNode or DominatorTreeNode as well.